xbmc/utils/UnicodeUtils.cpp

```diff
@@ -53,598 +53,699 @@
+  return EqualsNoCase(str1, str2, opt, Normalize);
 }
 
-bool UnicodeUtils::EqualsNoCase(const char *s1, const char *s2, StringOptions opt /* = StringOptions::FOLD_CASE_DEFAULT */,
-		const bool Normalize /* = false */) {
-	std::string str1 = std::string(s1);
-	std::string str2 = std::string(s2);
-	return EqualsNoCase(str1, str2, opt, Normalize);
+bool UnicodeUtils::EqualsNoCase(const char* s1,
+                                const char* s2,
+                                StringOptions opt / * = StringOptions::FOLD_CASE_DEFAULT * /,
+                                const bool Normalize / * = false * /)
+{
+  std::string str1 = std::string(s1);
+  std::string str2 = std::string(s2);
+  return EqualsNoCase(str1, str2, opt, Normalize);
 }
+*/
 
+int UnicodeUtils::Compare(const std::wstring_view& str1, const std::wstring_view& str2)
+{
 
-int UnicodeUtils::Compare(const std::wstring &str1, const std::wstring &str2) {
-
-	int32_t str1_length = str1.length();
-	int32_t str2_length = str2.length();
-	int32_t str1_start = 0;
-	int32_t str2_start = 0;
-	int rc = Unicode::StrCmp(str1, str1_start, str1_length,
-			                     str2, str2_start, str2_length);
-	return rc;
+  int32_t str1_length = str1.length();
+  int32_t str2_length = str2.length();
+  int32_t str1_start = 0;
+  int32_t str2_start = 0;
+  int rc = Unicode::StrCmp(str1, str1_start, str1_length, str2, str2_start, str2_length);
+  return rc;
 }
 
-int UnicodeUtils::Compare(const std::string &str1, const std::string &str2) {
+int UnicodeUtils::Compare(const std::string_view& str1, const std::string_view& str2)
+{
 
-	int32_t str1_length = str1.length();
-	int32_t str2_length = str2.length();
-	int32_t str1_start = 0;
-	int32_t str2_start = 0;
-	int rc = Unicode::StrCmp(str1, str1_start, str1_length,
-			                     str2, str2_start, str2_length);
-	return rc;
+  int32_t str1_length = str1.length();
+  int32_t str2_length = str2.length();
+  int32_t str1_start = 0;
+  int32_t str2_start = 0;
+  int rc = Unicode::StrCmp(str1, str1_start, str1_length, str2, str2_start, str2_length);
+  return rc;
 }
 
-int UnicodeUtils::CompareNoCase(const std::wstring &str1, const std::wstring &str2,
-		StringOptions opt /* = StringOptions::FOLD_CASE_DEFAULT */, const bool Normalize /* = false */) {
+int UnicodeUtils::CompareNoCase(const std::wstring_view& str1,
+                                const std::wstring_view& str2,
+                                StringOptions opt /* = StringOptions::FOLD_CASE_DEFAULT */,
+                                const bool Normalize /* = false */)
+{
 
-	int rc = Unicode::StrCaseCmp(str1, str2, opt, Normalize);
-	return rc;
+  int rc = Unicode::StrCaseCmp(str1, str2, opt, Normalize);
+  return rc;
 }
-int UnicodeUtils::CompareNoCase(const std::string &str1, const std::string &str2,
-    StringOptions opt /* = StringOptions::FOLD_CASE_DEFAULT */, const bool Normalize /* = false */) {
+int UnicodeUtils::CompareNoCase(const std::string_view& str1,
```

Comment:
a better step but using as reference with `view_string` types (`std::string_view&`) no makes sense, the container use a const pointer and little else,
but worse you prevent the use of view_string with char arrays / std::string types
which is one of the goal to remove superfluous overload methods

Just a side note, add the left `const` to `string_view` has no particular application, locks only the methods of the viewstring container, this because the view itself is always const/readonly and will never influence the origin of the data, then add `const` is fine only when you don't have to use the view methods. You will probably never use these methods since you make use of all customised string operation methods so its fine